lib/snowflex.ex

```diff
@@ -34,31 +34,29 @@ defmodule Snowflex do
 
   @type query_param :: {odbc_data_type(), [value()]}
   @type sql_data :: list(%{optional(String.t()) => String.t()})
-  @type query_opts :: [timeout: timeout(), map_null_to_nil?: boolean()]
+  @type connection_opts :: [timeout: timeout(), map_null_to_nil?: boolean()]
 
-  @spec sql_query(atom(), String.t(), query_opts()) ::
+  @spec do_query(pool_name :: atom(), query :: Query.t(), connection_opts()) ::
           sql_data() | {:error, term()}
-  def sql_query(pool_name, query, opts) do
+  def do_query(pool_name, query = %Query{params: nil}, opts) do
```

Comment:
Aren't `do_` functions at pepsico generally private functions?

Reply:
yes. I actually thought about moving all that logic into private functions in `Snowflex.Connection` or a new module and just leaving `Snowflex` empty or a place for exportable types. Thoughts?

Reply:
I'd leave snowflex as the public entry point.  Public types + public access to the functionality we are offering.